projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy_optimise.php

```diff
@@ -477,67 +507,77 @@ function merge_selectors(&$array) {
+			foreach ( $selectors as $s ) {
+				$simple_selectors = preg_split( '/\s*[+>~\s]\s*/', $s );
+				foreach ( $simple_selectors as $ss ) {
+					if ( $ss === '' ) {
 						$ok = false;
-					// could also check $ss for internal structure,
-					// but that probably would be too slow
+					}
+					// could also check $ss for internal structure, but that probably would be too slow.
 				}
 			}
-			if (!$ok)
-				unset($array[$selector]);
+			if ( ! $ok ) {
+				unset( $array[ $selector ] );
+			}
 		}
 	}
 
 	/**
 	 * Dissolves properties like padding:10px 10px 10px to padding-top:10px;padding-bottom:10px;...
-	 * @param string $property
-	 * @param string $value
+	 *
+	 * @param string $property - the property.
+	 * @param string $value - the value.
 	 * @return array
 	 * @version 1.0
 	 * @see merge_4value_shorthands()
 	 */
-	static function dissolve_4value_shorthands($property, $value) {
+	public static function dissolve_4value_shorthands( $property, $value ) {
 		$shorthands = & $GLOBALS['csstidy']['shorthands'];
-		if (!is_array($shorthands[$property])) {
-			$return[$property] = $value;
+		if ( ! is_array( $shorthands[ $property ] ) ) {
+			$return              = array();
+			$return[ $property ] = $value;
 			return $return;
 		}
 
 		$important = '';
-		if (csstidy::is_important($value)) {
-			$value = csstidy::gvw_important($value);
+		if ( csstidy::is_important( $value ) ) {
+			$value     = csstidy::gvw_important( $value );
 			$important = '!important';
 		}
-		$values = explode(' ', $value);
-
+		$values = explode( ' ', $value );
 
 		$return = array();
-		if (count($values) == 4) {
-			for ($i = 0; $i < 4; $i++) {
-				$return[$shorthands[$property][$i]] = $values[$i] . $important;
-			}
-		} elseif (count($values) == 3) {
-			$return[$shorthands[$property][0]] = $values[0] . $important;
-			$return[$shorthands[$property][1]] = $values[1] . $important;
-			$return[$shorthands[$property][3]] = $values[1] . $important;
-			$return[$shorthands[$property][2]] = $values[2] . $important;
-		} elseif (count($values) == 2) {
-			for ($i = 0; $i < 4; $i++) {
-				$return[$shorthands[$property][$i]] = (($i % 2 != 0)) ? $values[1] . $important : $values[0] . $important;
+		if ( count( $values ) === 4 ) {
+			for ( $i = 0; $i < 4; $i++ ) {
+				$return[ $shorthands[ $property ][ $i ] ] = $values[ $i ] . $important;
+			}
+		} elseif ( count( $values ) === 3 ) {
+			$return[ $shorthands[ $property ][0] ] = $values[0] . $important;
+			$return[ $shorthands[ $property ][1] ] = $values[1] . $important;
+			$return[ $shorthands[ $property ][3] ] = $values[1] . $important;
+			$return[ $shorthands[ $property ][2] ] = $values[2] . $important;
+		} elseif ( count( $values ) === 2 ) {
+			for ( $i = 0; $i < 4; $i++ ) {
+				$return[ $shorthands[ $property ][ $i ] ] = ( ( $i % 2 != 0 ) ) ? $values[1] . $important : $values[0] . $important; // phpcs:ignore Universal.Operators.StrictComparisons.LooseNotEqual
```

Comment:
No way for `$i % 2` to ever be a non-integer here.

Reply:
I was worried about floats here but looking again I don't think that's an issue.